core/src/main/java/org/mapfish/print/map/style/SLDParserPlugin.java

```diff
@@ -110,6 +120,18 @@ private String removeIndexReference(final String styleString) {
         try {
             reader = charSource.openBufferedStream();
 
+            // check if the XML is valid
+            // this is only done in a separate step to avoid that fatal errors show up in the logs
+            // by setting a custom error handler.
+            DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+            dbf.setNamespaceAware(true);
+            DocumentBuilder db = dbf.newDocumentBuilder();
+            db.setErrorHandler(new ErrorHandler());
+            db.parse(new InputSource(reader));
+            Closeables.close(reader, true);
```

Comment:
is there a potential resource leak on reader?  It seems to me that you should be using a `try {...}finally{...}` pattern with the Closer object to keep the code clean.  See [FileReportLoader.java](https://github.com/mapfish/mapfish-print/blob/4cbdc47c7e96ac47108806e8ea11d6da8bfcbdfc/core/src/main/java/org/mapfish/print/servlet/job/loader/FileReportLoader.java#L44) for an example